file/store.go

```diff
@@ -1,59 +1,120 @@
 package file
 
 import (
+	"crypto/rand"
+	"encoding/hex"
 	"io"
+	"os"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/s3crypto"
 )
 
+//go:generate moq -out filetest/uploader.go -pkg filetest . Uploader
+//go:generate moq -out filetest/cryptouploader.go -pkg filetest . CryptoUploader
+//go:generate moq -out filetest/vault.go -pkg filetest . VaultClient
+
+// Uploader represents the methods required to upload to s3 without encryption
+type Uploader interface {
+	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
+}
+
+// CryptoUploader represents the methods required to upload to s3 with encryption
+type CryptoUploader interface {
+	UploadWithPSK(input *s3manager.UploadInput, psk []byte) (*s3manager.UploadOutput, error)
+}
+
+// VaultClient is an interface to represent methods called to action upon vault
+type VaultClient interface {
+	WriteKey(path, key, value string) error
+}
+
 // Store provides file storage via S3.
 type Store struct {
-	config *aws.Config
-	bucket string
+	uploader       Uploader
+	cryptoUploader CryptoUploader
+	publicBucket   string
+	privateBucket  string
+	vaultPath      string
+	vaultClient    VaultClient
 }
 
 // NewStore returns a new store instance for the given AWS region and S3 bucket name.
-func NewStore(region string, bucket string) *Store {
+func NewStore(region string, publicBucket, privateBucket, vaultPath string, vaultClient VaultClient) *Store {
 
 	config := aws.NewConfig().WithRegion(region)
 
+	session, err := session.NewSession(config)
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
```

Comment:
Do we want the app to exit here? I assume this is called from `main` on start up. IMO it would be cleaner to return the error to main and let that exit the app.